mautic/mautic
# Add empty and not empty to segment stage filters.

Add empty and not empty to segment stage filters. Also fixes a bug with eq/neq.

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Yes
| New feature? |  Yes
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/2485
| BC breaks? | Not Sure (what is BC?)
| Deprecations? | No

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

Adds empty and not empty to stages in segment filters.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:

#### Steps to test this PR:
1. Create a contact and make sure stage is null / not set.
2. Create a segment where stage is set to empty.
3. Run `php app/console mautic:segments:rebuild` and make sure only 1 contact is added to segment.
4. Set the contact's stage to something not null.
5. Modify the segment to filter by stage not empty.
6. Run the segment rebuild again and make sure the contact is back in the segment.

#### List deprecations along with the new alternative:
1. 
2. 

#### List backwards compatibility breaks:
1. 
2.
